### ISSUE TITLE
[MU4 Task] Restore MIDI import panel

### ISSUE THREAD

**Issue by shoogle** [musescore/MuseScore#11918]
**Task description**

Port the MIDI Import panel to MU4, either as a Qt Widget or as a QML panel.

![image](https://user-images.githubusercontent.com/11011881/172492749-b62069b4-9f82-4b44-8a09-8c48b4568f6c.png)

The MIDI Import panel is displayed in when opening a MIDI file in MU3 but not yet in MU4. In MU3 it appears at the bottom of the screen and contains options to customize the MIDI import.

**Comment by shoogle** [comment]
Since this is potentially quite a big task, I've temporarily assigned to @Tantacrul for prioritisation.

**Comment by shoogle** [comment]
@Tantacrul says: if the MIDI Import panel cannot be easily ported as a Qt Widget then it can be delayed until 4.x.

**Comment by RomanPudashkin** [comment]
> @Tantacrul says: if the MIDI Import panel cannot be easily ported as a Qt Widget then it can be delayed until 4.x.

Yes, unfortunately, it cannot be easily ported to MU4. I'll move this task to 4.x

@Tantacrul FYI

**Comment by Tantacrul** [comment]
Thanks. Agreed.

**Comment by zDEFz** [comment]
Today I did to transform a short youtube snippet into midi with a converter, which succeeded.
However, MuseScore 4.2 does not match the timing properly.
Having some options to comply to the timing m

Heres something you can test with:

[Archive.zip](https://github.com/musescore/MuseScore/files/13720664/Archive.zip)
ore would be nice, given the possibility.

**Comment by Jojo-Schmitz** [comment]
Why is this closed?

**Comment by cbjeukendrup** [comment]
That still dates from the "we close issues that we postpone to 4.x" era

**Comment by Jojo-Schmitz** [comment]
Which really was a bad idea...

**Comment by margaridamcraveiro** [comment]
My colleague @cheilagoncalves and I would really like to work on this issue, as well as maybe talk with developers about improving the UI.

**Comment by Tantacrul** [comment]
> My colleague [@cheilagoncalves](https://github.com/cheilagoncalves) and I would really like to work on this issue, as well as maybe talk with developers about improving the UI.

@cbjeukendrup - perhaps we can discuss facilitating this when you have time? I can certainly help with the UI improvements.

**Comment by margaridamcraveiro** [comment]
@Tantacrul @cbjeukendrup We can start by adding the menu, if you think it's ok, and later improve the UI.

**Comment by cbjeukendrup** [comment]
@Tantacrul @margaridamcraveiro @cheilagoncalves It might be a good idea to have a chat on Discord first, to make sure this project will go well. I'll send you a direct message.

**Comment by Tantacrul** [comment]
I definitely think it'd be better to have just a quick concept for how we want it to be better. I'm not so worried about cosmetics as I am UX. For example - where is the option kept? How is it triggered? Should there be one or two steps to make the whole process easier to understand? etc. This can be very roughly sketched out, so you can get to work quickly.

I'll need to have a look at the old version and then remind myself about how it works. Happy to also chat about it quickly on Discord and we can just hash out the details. I'm sure it'll be reasonably quick to do so.

**Comment by lgmventura** [comment]
Would be great to leave some space in terms of flexibility of expandable features (to keep simplicity in a first glance) to add future options, like automatically adding dynamic marks for velocity changes, what per se would require more options, like thresholds, including hairpins for continuous changes. Other options could come in the future as well, like trying to identify ornaments from the MIDI data, accents, differentiation of staccato and stacatíssimo...
There is also room for a lot of configuration tweaks when it is marked as human performance.

**Comment by Jojo-Schmitz** [comment]
Came up again in https://musescore.org/en/node/386806

**Comment by Jojo-Schmitz** [comment]
Came up again in https://musescore.org/de/node/390915

**Comment by andrewwyld** [comment]
I would appreciate some means of directly cleaning up a MIDI import. For instance, I have found that ctrl/cmd-shift-up/down can affect which staff a note *appears* on, but not which part it belongs to.

For instance, I have a part where a single quaver in the left hand is followed by a quaver melody in the right hand, in a way that could reasonably appear like a single quaver phrase. I did ctrl/cmd-shift-down on the first note to move it to the left staff, and the displayed part was still connected to the RH quavers by a beam.

<img width="370" height="185" alt="Image" src="https://github.com/user-attachments/assets/0f7b28bb-d74f-448d-91ed-ca772b60b8b8" />

However, cutting this note:

<img width="351" height="173" alt="Image" src="https://github.com/user-attachments/assets/abdab176-0e92-4efc-b370-beabdeb81c5e" />

and pasting it over this quaver rest:

<img width="349" height="178" alt="Image" src="https://github.com/user-attachments/assets/91d72cbd-42ac-407a-850a-f4bf35456182" />

gave the desired appearance:

<img width="355" height="193" alt="Image" src="https://github.com/user-attachments/assets/0e3b3893-387b-4e79-b2b9-886a54fb8a1c" />

I can only assume that the quaver rest over which I pasted the part "belonged" to the left hand and in some way I therefore reassigned the note to the left hand part. A means of doing this directly and under control would be great. In fact, I think when a user performs ctrl/cmd-shift-up/down, this is probably what they normally *want* to do. In particular, the above method doesn't work where two notes of a right hand chord have been split between left and right hands; the paste will overwrite the other note in the chord. Having the ctrl/cmd-shift-up/down behaviour, by default, shift the note to the left- or right-hand *part* rather than merely the staff would be a major improvement.

**Comment by MarcSabatella** [comment]
The commands you refer to are specifically meant for creating cross staff notation, and it is correct and necessary for this purpose that the notes remain on their originally staves.  What you are describing would be a  *new* command intended for a totally different purposes.  Feel free to own a new feature request specifically for that.
